### DISCUSSION TITLE
How to add a label like a control to an object that will also display during multiselection?

### DISCUSSION THREAD

**Discussion by oldben87** [fabricjs/fabric.js#8848] [Q&A]
I am looking to display an object on a canvas that has a label next to it to indicate which item it is.

I have tried adding this label as a custom control, which works nicely for different scale and zoom levels.
However during multi selection, the control disappears off of the object. And you could have several objects selected and forgot which one is which.

Is there a way to enable a custom control to show on individual objects during multi selection?

**Comment by ShaMan123** [comment]
I have been hearing a lot about this need and I use it as well.
You can override the relevant place.
Or `_renderControls` on canvas or in ActiveSelection if I am not mistaken

**Reply by ShaMan123** [reply]
@asturur I think we should expose a normal way to easily configure this

**Reply by oldben87** [reply]
Thanks for the speedy response!

I see that if I pass in a {hasControls: true} object as the third argument to the activeSelection._renderControls method then it is now showing all the controls for the object. Is this what you meant to do?

**Comment by ShaMan123** [comment]
Yep

בתאריך יום ה׳, 20 באפר׳ 2023, 21:18, מאת oldben87 ***@***.***
